araproc/analysis/snr.py

```diff
@@ -9,16 +10,21 @@ def get_vpp(waveform):
 
     Parameters
     ----------
-    waveform: TGraph
-        A TGraph of the waveform.
+    waveform: TGraph or np.ndarray
+        A TGraphi or np.ndarray of the waveform voltage.
 
     Returns
     -------
     vpp : float
         Peak-to-peak voltage in same units as trace.
     """ 
 
-    _, trace = wfu.tgraph_to_arrays(waveform)
+    if(isinstance(waveform, ROOT.TGraph)):
+      _, trace = wfu.tgraph_to_arrays(waveform)
+    elif(isinstance(waveform, np.ndarray)):
```

Comment:
I actually think you want to be more aggressive than this. You don't just care that it's an ndarray. For this to make sense, it needs to be an ndarray of a specific shape, in particular it needs to be a 1D array. So I'd add enforcement for that.

Reply:
Okay how's that?